### PR TITLE
MAC address cleanup

### LINKED ISSUE #199: MAC address conflict when restoring a virtual machine to alternate namespace
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
> /kind bug
/kind enhancement
/remove-lifecycle rotten

**What happened**:

When a virtual machine is restored to an alternate namespace, the virtual machine is restored with the same MAC address as the original virtual machine.  This results in a MAC address conflict if the original virtual machine is still running on the original namespace. 

**What you expected to happen**:

Provide a way to blank the MAC address on restore. 

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

The issue was resolved by updating the plugin to clear the MAC addresses on the restore item action

**Environment**:
- CDI version (use `kubectl get deployments cdi-deployment -o yaml`):
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by alromeros**
**What this PR does / why we need it**:

When a virtual machine is restored to an alternate namespace, the virtual machine is restored with the same MAC address as the original one. This results in a MAC address conflict if the original virtual machine is still running on the original namespace.

Though we already have a work-in-progress PR, recent interest in this feature has forced us to open a new one to address this issue. From now on, users can optionally clean a VM or VMI MAC address by setting the `"velero.kubevirt.io/clear-mac-address"` label in the restore object.

Cherry-picked a couple of commits from https://github.com/kubevirt/kubevirt-velero-plugin/pull/235 to keep the original contributor's work.

/cc @esteban-ee

**Which issue(s) this PR fixes**:
Fixes #199

**Release note**:

```release-note
Support clearing MAC address by label
```

### CODE REVIEW

File: pkg/plugin/vm_restore_item_action.go
Comment (ShellyKa13) [inline pkg/plugin/vm_restore_item_action.go]: dont think you need this commit..

File: pkg/plugin/vm_restore_item_action.go
Comment (ShellyKa13) [inline pkg/plugin/vm_restore_item_action.go]: typo Adress -> address
Also I think the name is missleading.. 
consider something like: ShouldClearMacAddress

File: pkg/plugin/vm_restore_item_action.go
Comment (ShellyKa13) [inline pkg/plugin/vm_restore_item_action.go:70]: I think its better also to put this in a function ClearMacAddress

File: pkg/util/util_test.go
Comment (ShellyKa13) [inline pkg/util/util_test.go]: the name of the tests should also explain what is tested i.e here: with no annotation, in the next with annotation false etc..

File: pkg/util/util_test.go
Comment (ShellyKa13) [inline pkg/util/util_test.go:428]: you should add here "without label" and in the next test "with label"

File: tests/vm_backup_test.go
Comment (ShellyKa13) [inline tests/vm_backup_test.go:223]: maybe you should use patch here?

File: tests/vm_backup_test.go
Comment (ShellyKa13) [inline tests/vm_backup_test.go:237]: why do you need to stop the vm before you delete it?

**Review by ShellyKa13 — COMMENTED** [review]
Some comments are relevant to the 2 commit which are not yours.. I get the desire to use community member commits but they dont have much value.. I believe you should either delete them or squash the first 3 commits and add him as as signed by also..

File: pkg/plugin/vm_restore_item_action.go
Comment (alromeros) [reply pkg/plugin/vm_restore_item_action.go]: I'm not against having "incomplete" commits such as these if they showcase the development history and specifically wanted to make an exception for these as they are from another contributor. That said, we've had this conversation in the past and I'm also ok with a cleaner approach, so I'll squash them as suggested.

File: pkg/plugin/vm_restore_item_action.go
Comment (alromeros) [reply pkg/plugin/vm_restore_item_action.go]: Related to my comment above

File: tests/vm_backup_test.go
Comment (alromeros) [reply tests/vm_backup_test.go:237]: Just a remnant from copy-pasting another test that does the same. I can remove it.

File: tests/vm_backup_test.go
Comment (alromeros) [reply tests/vm_backup_test.go:223]: This is a preexisting function used in several other tests, prefer to leave it here and consider refactoring the function itself in a follow-up.

**Review by awels — APPROVED** [review]
putting hold on it so Shelly can take a look, but if not before you want it merged, you have the lgtm
/hold

**Comment by ShellyKa13** [comment]
/unhold
/lgtm
Nice work! thanks!

**Comment by alromeros** [comment]
Will backport this once https://github.com/kubevirt/kubevirt-velero-plugin/pull/310 is meged, thanks @ShellyKa13!

**Pull request merged**
